lib/ansible/playbook/role/metadata.py

```diff
@@ -101,10 +101,16 @@ def _load_dependencies(self, attr, ds):
             if 'ansible.legacy' not in collection_search_list:
                 collection_search_list.append('ansible.legacy')
 
+        # The dependency format is used by ansible-galaxy and when running roles.
+        # Since role runtime (with argument spec) requires all role params be documented,
+        # add a supplemental spec so this isn't a failure by default for ansible-galaxy options.
+        # If the role actually documents these options, they'll be validated normally.
+        internal_galaxy_spec = {k: {} for k in VALID_SPEC_KEYS}
```

Comment:
i would probably just remove the full key and add a debug with that info, for those that are 'overloading' role params and expecting a variable to pass through. I expect 'version' to be widely used (as 'name' was before).

Reply:
i just remembered, we did add that warning at one point .. .i'm just not sure where/at what point

Reply:
I feel like we'd need to deprecate this dependencies syntax for ansible-galaxy if we want to issue a warning at runtime by default. I could add a debug message in addition to or instead of the vvv message in validate_argument_spec.